### PR TITLE
rgw container options

### PR DESCRIPTION
**Pull request by leseb**
Now, we will pass the right flag in the container startup. So for
example radosgw_civetweb_options option will be exposed in the rgw
container.
Also this PR is the foundation to support multiple backend, such as the
new 'beast' from Ceph Mimic.

Closes: https://bugzilla.redhat.com/show_bug.cgi?id=1582411
Signed-off-by: Sébastien Han <seb@redhat.com>

### CODE REVIEW

**Comment by leseb** [comment]
jenkins test docker_cluster

**Comment by leseb** [comment]
jenkins test luminous_docker_cluster

**Comment by leseb** [comment]
jenkins test luminous-docker_cluster

File: roles/ceph-defaults/defaults/main.yml [hunk 1]
```diff
@@ -381,12 +381,21 @@ mds_max_mds: 1
...
+radosgw_civetweb_options: "num_threads={{ radosgw_civetweb_num_threads }}"
 # For additional civetweb configuration options available such as SSL, logging,
 # keepalive, and timeout settings, please see the civetweb docs at
 # https://github.com/civetweb/civetweb/blob/master/docs/UserManual.md
-radosgw_civetweb_options: "num_threads={{ radosgw_civetweb_num_threads }}"
+
+radosgw_beast_options:
+
+radosgw_fronted_options: "{{ radosgw_civetweb_options if radosgw_fronted_type == 'civetweb' else radosgw_beast_options }}"
+radosgw_fronted_port: "{{ radosgw_civetweb_port }}"
```
Comment (leseb) [inline roles/ceph-defaults/defaults/main.yml]: double check this

**Comment by leseb** [comment]
jenkins test pipeline

**Comment by leseb** [comment]
CI failure is unrelated. Just ceph-ansible-prs-luminous-bluestore_osds_container failed, which does not deploy any rgw.

**Comment by guits** [comment]
unrelated failure, merging.

**Pull request merged**